### ISSUE TITLE
Add option to use file with environment variables in new-app

### ISSUE THREAD

**Issue by ppitonak** [openshift/origin#7585]
It is possible to specify environment variables directly to `oc new-app` command

```
$ oc new-app openshift/postgresql-92-centos7 \
-e POSTGRESQL_USER=user \
-e POSTGRESQL_DATABASE=db \
-e POSTGRESQL_PASSWORD=password
```

Add an option to specify a file with env variables so that they don't have to be specified on command line (was implemented in OpenShift 2, Docker has --env-file option)

```
$ oc new-app openshift/postgresql-92-centos7 --env-file /tmp/myenvironmen
```

where file would contain

```
POSTGRESQL_USER=user
POSTGRESQL_DATABASE=db
POSTGRESQL_PASSWORD=password
```

**Comment by mfojtik** [comment]
@ppitonak sounds reasonable to me. @smarterclayton ?

**Comment by fabianofranz** [comment]
Yes. In `rhc create-app` you could either pass `KEY=VALUE` *or* the path to a file using the same `--env` flag, which is also completely feasible. No strong preference.

**Comment by smarterclayton** [comment]
It's reasonable.  Note that for new-app you don't even need `-e`.  You can do

    oc new-app POSTGRES_USER=hello ... openshift/postgresql-92-centos7

**Comment by bparees** [comment]
i think we should split the new-app updates out of this PR so we can move forward w/ the api+parameter substitution changes, regardless of what we decide to do w/ the new-app output.

that way @jwforres can pick up this change for the web console sooner than later.

**Comment by ravisantoshgudimetla** [comment]
@bparees : I wish to contribute to this PR, I am new to Openshift and need you guys help. Let me know, if I can pick this up.

**Comment by bparees** [comment]
@ravisantoshgudimetla best place to start would be our contributor guide:
https://github.com/openshift/origin/blob/master/CONTRIBUTING.adoc

once you have a dev environment up and running we can discuss how to solve this issue.

**Comment by ravisantoshgudimetla** [comment]
@bparees I have set the dev environment up. I am currently reading newapp.go in <home>/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd.

**Comment by bparees** [comment]
@ravisantoshgudimetla ok, that's the right place to start, i think the original issue summary describes the intent pretty well, so I guess just let me know where you have questions.

**Comment by ravisantoshgudimetla** [comment]
@bparees Cool, I will go through it and get back to you, in case I have questions. Thanks.

**Comment by ravisantoshgudimetla** [comment]
@bparees - Following are the set of changes that I thinking of making for this PR. The idea is to reuse the existing code as much as possible. I am trying to parse the environment file and get the []string from it and append it to environment []string in newapp.go .

 cmd.Flags().StringSliceVarP(&config.EnvironmentFile, "envfile", “", config.Environment, “Specify location for key-value pair environment file”) in newapp.go in origin/pkg/cmd/cli/cmd
In origin/pkg/generate/app/cmd/newapp.go in AddArguments function following line is going to be added.
   case app.IsPossibleEnvironmentFile(s): 
c.Environment = append(c.Environment, app.getEnvironmentVarList(s)) //This is easiest but probably not consistent with whats already there.
In origin/pkg/generate/app/environment.go file is created, which has IsPossibleEnvironmentFile() which takes the file path and checks if its a file and getEnvironmentVarList which takes the file and reads the values in it line by line, have some validation to see if its of format key-value pair and return the []string which can be appended in c.Environment.

Please let know, if I am missing something or if there is an easier way to do this.

**Comment by bparees** [comment]
@ravisantoshgudimetla generally sounds reasonable, go ahead and open the PR and we'll continue the discussion from there.

**Comment by ravisantoshgudimetla** [comment]
@bparees, as discussed raised PR. Please note that I haven't written the test cases yet. Once the approach seems reasonable , I will go ahead and write some test cases.

**Comment by jtslear** [comment]
I know I'm horribly late to the game, but what if we simply improved the documentation, and relied on shell utilities to complete?
`oc new-app $(cat test.env) --docker-image=jtslear/testapp`

**Comment by bparees** [comment]
@mmilata the card you're working this sprint should resolve this.

**Comment by mmilata** [comment]
Should be fixed in #12164.
